geomstats/geometry/so3.py

```diff
@@ -0,0 +1,1664 @@
+"""The special orthogonal group SO(n).
+
+i.e. the Lie group of rotations in n dimensions.
+"""
+
+import geomstats.backend as gs
+from geomstats.geometry.embedded_manifold import EmbeddedManifold
+from geomstats.geometry.general_linear import GeneralLinear
+from geomstats.geometry.lie_group import LieGroup
+from geomstats.learning.frechet_mean import FrechetMean
+
+ATOL = 1e-5
+
+TAYLOR_COEFFS_1_AT_0 = [1., 0.,
+                        - 1. / 12., 0.,
+                        - 1. / 720., 0.,
+                        - 1. / 30240., 0.]
+TAYLOR_COEFFS_2_AT_0 = [1. / 12., 0.,
+                        1. / 720., 0.,
+                        1. / 30240., 0.,
+                        1. / 1209600., 0.]
+TAYLOR_COEFFS_1_AT_PI = [0., - gs.pi / 4.,
+                         - 1. / 4., - gs.pi / 48.,
+                         - 1. / 48., - gs.pi / 480.,
+                         - 1. / 480.]
+
+
+class SpecialOrthogonal(LieGroup, EmbeddedManifold):
+    """Class for the special orthogonal group SO(n).
+
+    i.e. the Lie group of rotations.
+    """
+
+    def __init__(self, n, point_type=None, epsilon=0.):
+        """Initialize an instance of SO(n).
+
+        Parameters
+        ----------
+        n : int
+            the dimension of the euclidean space that SO(n) acts upon
+        point_type : str, {'vector', 'matrix'}, optional
+            if None is given, point_type is set to 'vector for dimension 3
+            and matrix otherwise
+        epsilon : float, optional
+            precision to use for calculations involving potential divison by in
+            rotations
+            default: 0
+        """
+        assert isinstance(n, int) and n > 1
```

Comment:
This whole file is in fact the copy-paste of son, right? We should adapt and change n to 3, otherwise, users will be very confused.

Reply:
A of doing it would be:
- rename `SpecialOrthogonal` to `SpecialOrthogonal3` and have it inherit from `SpecialOrthogonal`
- Remove `n` parameter in __init__
- set `self.n = 3`
- Delete methods that do not explicitly require the 3D, so that the parent's methods are called.